### PR TITLE
Consider overrides when checking if an environment satisfies requirements

### PR DESCRIPTION
**Pull request by zanieb**
I noticed this while implementing #4973 — it seems like an oversight.

### CODE REVIEW

File: crates/uv/src/commands/pip/install.rs [hunk 1]
```diff
@@ -176,7 +176,7 @@ pub(crate) async fn pip_install(
     // Ideally, the resolver would be fast enough to let us remove this check. But right now, for large environments,
     // it's an order of magnitude faster to validate the environment than to resolve the requirements.
     if reinstall.is_none() && upgrade.is_none() && source_trees.is_empty() && overrides.is_empty() {
```
Comment (ibraheemdev) [inline crates/uv/src/commands/pip/install.rs:178]: Looks like `overrides` is empty here?

Comment (charliermarsh) [reply crates/uv/src/commands/pip/install.rs:178]: Yeah I think `satisfies` doesn't quite work with overrides as-is.

File: crates/uv-installer/src/site_packages.rs [hunk 2]
```diff
@@ -319,6 +320,22 @@ impl SitePackages {
                             RequirementSatisfaction::Satisfied => {}
                         }
                     }
+                    // Validate that the installed version satisfies the overrides.
```
Comment (charliermarsh) [inline crates/uv-installer/src/site_packages.rs:323]: I think we'd need to change the code such that all _dependencies_ also respect overrides.

Comment (zanieb) [reply crates/uv-installer/src/site_packages.rs:323]: Yeah I noticed this in my test at https://github.com/astral-sh/uv/pull/4973/files#diff-a7df5406f8301d2dd603b5355f282be5afa0198ff71f03144ab6d8aed56c07a3R713-R719 — I'll explore fixing this before merging here. I'm not sure how I'll write a nice test case though without the `uv run` changes.

**Comment by zanieb** [comment]
It turns out this is quite complicated because overrides can be unnamed which isn't compatible with the current design of `satisfies` — we cannot check that the override is respected without significant changes.

It also looks like there's a bug in the constraint enforcement in this method, where it does not check that the constraint name matches the installed distribution name.

**Pull request closed without merging**